### ISSUE TITLE
Implement caching

### ISSUE THREAD

**Issue by AWhetter** [readthedocs/sphinx-autoapi#191]
Currently AutoAPI will generate the documentation from scratch every time. It would be useful to have a way to get it to not regenerate the documentation if the source code being documented has not changed.
If necessary, it's fine for this mode to need `autoapi_keep_files` to be turned on.

**Comment by potiuk** [comment]
I hope it's not April's fool kind of issue. We'd love to get it in Airflow :)

**Comment by lalten** [comment]
Hey @AWhetter, I would love to have a caching feature in AutoAPI! Any chance to reopen this?

My use case is that I am integrating Sphinx into a very large Bazel project. Currently if any single file changes, the entire documentation needs to build for many minutes.
I have tried to run a minimal Sphinx+AutoAPI on every single source as Bazel Aspect, declare the rendered rst files as Aspect outputs (which makes Bazel cache them nicely), and feed them to Sphinx in a final step to assemble the HTML. However (a little too late) I understood that the rst files don't contain all necessary information, and AutoAPI will fail to follow their directives (KeyErrors) because it does not have the internal mapping representations loaded.

Maybe AutoAPI could place the pickled mappings next to the rendered rst files; and then load them later to avoid re-parsing original sources?

On a related note, maybe this would be a nice opportunity to split that first part into a standalone, sphinx-free library / application.

Let me know what you think!

**Comment by AWhetter** [comment]
The feature was implemented and release in v1.6.0.

**Comment by potiuk** [comment]
cc @mik-laj

**Comment by mik-laj** [comment]
@potiuk I saw it, but our project (apache/airflow) is unfortunately not compatible with the newer versions of sphinx-autoapi.  astroid cannot parse the source code. We need to use sphinx-autoapi==1.0.0
https://github.com/apache/airflow/blob/5a29e12f8a1ab104301aebf60969fdb0adfb05e7/constraints-3.8.txt#L367

There are many bugs, but one of which I was able to detect was a failure to process this line.
https://github.com/apache/airflow/blob/10b8ecc86f24739a38e56347dcc8dc60e3e43975/airflow/stats.py#L349
But then there are more errors that I didn't diagnose
